src/lib/falcor/routes/articles/database-calls.js

```diff
@@ -20,3 +24,67 @@ export async function getPaginatedArticle(database, pageLength, pageIndex) {
     .offset(offset);
   return articles;
 }
+
+/**
+ * Updates the tags for a given article.
+ * @param   {database} database    The database
+ * @param   {int}      articleId   The id of the article.
+ * @param   {int[]}    newTags     The ids of the tags.
+ * @returns {string[]} slugs       The edited slugs.
+ */
+export async function updateArticleTags(database, articleId, newTags) {
+  // Delete old tags.
+  await database('articles_tags')
+    .where('article_id', '=', articleId)
+    .del();
+
+  // Insert new tags.
+  const insertArray = _.map(newTags, tagId => ({
+    article_id: articleId,
+    tag_id: tagId,
+  }));
+  await database('articles_tags').insert(insertArray);
+
+  // Get updated slugs.
+  const rows = await database
+    .select('slug')
+    .from('articles_tags')
+    .innerJoin('tags', 'tags.id', '=', 'articles_tags.tag_id')
+    .where('article_id', '=', articleId);
+  return rows.map(row => row.slug);
+}
+
+/**
+ * @param   {database} database    The database
+ * @param   {string[]} slugs       The slugs whose tags we receive
+ * @returns {Object}   data        The data
+ */
+export async function articleTagQuery(database, slugs) {
+  // slugs function parameter is an array of article slugs
+  // of which to fetch the tag of.
+  // The function returns an object with article slugs
+  // as keys and values being arrays of author slugs.
+  const rows = await database
+    .select('articles.slug as articleSlug', 'tags.slug as tagSlug')
+    .from('tags')
+    .innerJoin('articles_tags', 'tags.id', '=', 'tag_id')
+    .innerJoin('articles', 'articles.id', '=', 'article_id')
+    .whereIn('articles.slug', slugs)
+    .orderBy('articles_tags.id', 'asc')
+    .catch(e => {
+      // database.destroy();
+      throw new Error(e);
```

Comment:
It automatically throws/rejects when using async / await, so catch isn't needed here